src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFileUtil.java

```diff
@@ -97,12 +98,12 @@ public FilePath call() throws IOException {
 
             FilePath target = null;
             if (createTempFile) {
-                target = ManagedFileUtil.createTempFile(build.getWorkspace().getChannel());
+                target = ManagedFileUtil.createTempFile(workspace.getChannel());
             } else {
                 
                 String expandedTargetLocation = managedFile.targetLocation;
                 try {
-                    expandedTargetLocation = TokenMacro.expandAll(build, listener, managedFile.targetLocation);
+                    expandedTargetLocation = build instanceof AbstractBuild ? TokenMacro.expandAll((AbstractBuild<?, ?>) build, listener, managedFile.targetLocation) : managedFile.targetLocation;
```

Comment:
> It would be useful to have a "Token macro" expression support in workflow.

No, this is expected to be handled by the Workflow Groovy script itself. Whatever values you finally provide to the step should be used as is.